x-pack/plugin/repository-encrypted/src/main/java/org/elasticsearch/repositories/encrypted/EncryptedRepository.java

```diff
@@ -637,7 +637,7 @@ public void writeBlob(String blobName, InputStream inputStream, long blobSize, b
                 InputStream encryptedInputStream = ChainingInputStream.chain(
                     dekIdBytes.streamInput(),
                     new EncryptionPacketsInputStream(
-                        inputStream,
+                        Streams.noCloseStream(inputStream),
```

Comment:
I suppose we could also fix this in `EncryptionPacketsInputStream` but this seemed less trappy and easier to reason about for now.